tests/test_inference.py

```diff
@@ -774,6 +774,36 @@ def test_three_sites(self):
         self.verify([[1, 1], [1, 1], [1, 1]])
 
 
+class TestTimeUnits:
+    """
+    Tests that time units get set properly
+    """
+
+    def test_normally_uncalibrated(self):
+        ts = msprime.simulate(10, mutation_rate=5, recombination_rate=4, random_seed=21)
```

Comment:
If we're not testing the data here I don't see any value in calling ``msprime.simulate``. We're just adding extra complexity here from the effects of ``from_tree_sequence`` which is very complicated and isn't the actual path that users will use. It would be better to have a super-simple minimal example, and check the individual steps in isolated tests.

Probably good to test from_tree_sequence by itself, as I guess that may inherit the time_units in different ways as well.